src/mender-update/daemon/context.cpp

```diff
@@ -373,10 +435,16 @@ static error::Error UnmarshalJsonStateData(const json::Json &json, StateData &st
 	auto exp_int = json.Get("Version").and_then(json::ToInt);
 	SetOrReturnIfError(state_data.version, exp_int);
 
-	if (state_data.version != kStateDataVersion) {
+	if (state_data.version != kStateDataVersion && state_data.version != 1
+		&& state_data.version != 0) {
```

Comment:
No, zero is invalid, because we started from one. And it could indicate a Golang struct which is improperly initialized, for example, so should definitely be avoided.